examples/simple/foo_library/foo_library.cc

```diff
@@ -13,21 +13,30 @@ nostd::shared_ptr<trace::Tracer> get_tracer()
 
 void f1()
 {
-  auto span = get_tracer()->StartSpan("f1");
+  auto span  = get_tracer()->StartSpan("f1");
```

Comment:
Consider a convenient method `StartAsActiveSpan` https://github.com/open-telemetry/opentelemetry-python/blob/23cf584f1016146877482daecbc92a4acd546623/docs/examples/basic_tracer/basic_trace.py#L13.

Reply:
Yes, I thought about that. For that to work, we'd need to make `StartAsActiveSpan` return a `Scope`, and add a `GetSpan` to the `Scope`.

If we want to do that I'd like to have a separate PR for that, and first clarify if the `Scope` solution is generally viable.

Reply:
Sounds good. Not a blocker for this PR.